src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs

```diff
@@ -169,6 +163,13 @@ private bool UsePlaceholderChannel(string workerRuntime)
                     // Use if readonly and not v2 compatible on ~3 extension
                     return _applicationHostOptions.CurrentValue.IsFileSystemReadOnly && !_environment.IsV2CompatibileOnV3Extension();
                 }
+
+                // If a profile evaluates to true and was not previously loaded, restart worker process
+                if (!_profileManager.IsCorrectProfileLoaded(workerRuntime))
+                {
```

Comment:
Is the ordering of the checks in this method important? If not, wondering if it would make sense for this condition to be added to the check in line 151 to avoid duplication of "return false" ?

Reply:
I'm not 100% sure, but I think ordering might be important here. @fabiocav & @pragnagopa probably know more about this code path

Reply:
Not sure about the ordering. I felt it was better to respect previous conditions to not use placeholder channel over the newly introduced profiles change.